src/hooks/twilio.ts

```diff
@@ -208,49 +206,51 @@ export const useVideoRoomState = ({
     hide: turnVideoOff,
   } = useShowHide(showVideoByDefault);
 
-  const localParticipant = room?.localParticipant;
-
-  useEffect(() => {
-    if (!localParticipant) return;
-
-    setParticipants((prevParticipants) => [
-      // Hopefully prevents duplicate users in the participant list
-      ...prevParticipants.filter(
-        (p) => p.identity !== localParticipant.identity
-      ),
-      localParticipant,
-    ]);
-  }, [localParticipant]);
-
-  useEffect(() => {
-    if (!token || !roomName) return;
-
-    const participantConnected = (participant: RemoteParticipant) => {
-      setParticipants((prevParticipants) => [...prevParticipants, participant]);
-    };
+  const participantConnected = useCallback((participant: RemoteParticipant) => {
+    setParticipants((prevParticipants) => [...prevParticipants, participant]);
+  }, []);
 
-    const participantDisconnected = (participant: RemoteParticipant) => {
+  const participantDisconnected = useCallback(
+    (participant: RemoteParticipant) => {
       setParticipants((prevParticipants) =>
         prevParticipants.filter((p) => p !== participant)
       );
-    };
+    },
+    []
+  );
+
+  useEffect(() => {
+    if (!token || !roomName) return;
 
     // https://media.twiliocdn.com/sdk/js/video/releases/2.7.1/docs/global.html#ConnectOptions
     connect(token, {
       name: roomName,
       video: hasVideo,
       enableDscp: true,
-    }).then((room) => {
-      setRoom(room);
-      room.on("participantConnected", participantConnected);
-      room.on("participantDisconnected", participantDisconnected);
-      room.participants.forEach(participantConnected);
-    });
+    }).then(setRoom);
 
     return () => {
       disconnect();
     };
-  }, [disconnect, roomName, token, hasVideo]);
+  }, [
+    disconnect,
+    roomName,
+    token,
+    hasVideo,
+    participantConnected,
+    participantDisconnected,
+  ]);
```

Comment:
Looks like this dependency array wasn't updated? I believe it can be way less than this now?

Reply:
Fixed in https://github.com/sparkletown/sparkle/pull/1299/commits/ce562c6a4616118ea904bb8f088ca751787a5f95